From a4c89218f2ad39d24953122bec80469d40d7141e Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Thu, 25 Oct 2012 15:09:50 +0000
Subject: [PATCH 0359/1076] Mantis 0024819: mcontrol: illogical test for boot
 time: hour >= boot.hour _and_ minute >= boot.minute check
 !pServant->isKioskBusy()

git-svn-id: svn://localhost/SelfServiceCommon/trunk@428 90b65887-3827-0410-9a23-83215b262276
---
 SelfServiceCommon/Massai/cpp/Controller/src/reboot.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/SelfServiceCommon/Massai/cpp/Controller/src/reboot.cpp b/SelfServiceCommon/Massai/cpp/Controller/src/reboot.cpp
index 9fa23cf0..2a18e80d 100644
--- a/SelfServiceCommon/Massai/cpp/Controller/src/reboot.cpp
+++ b/SelfServiceCommon/Massai/cpp/Controller/src/reboot.cpp
@@ -133,7 +133,9 @@ long checkReboot::function()
 
         unsigned const minutes_of_reset = ( lHourReset * 60 ) + lMinuteReset + lDelay;
         unsigned const minutes_of_now = ( time.hours() * 60 ) + time.minutes();
-        if(!fRebootedToday && (minutes_of_now >= minutes_of_reset ))
+        if(!fRebootedToday
+          && !pServant->isKioskBusy()
+          && (minutes_of_now >= minutes_of_reset))
         {
           rc = pServant->rebootKiosk();
 
-- 
2.41.0.windows.1

